core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -523,6 +523,16 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def copyFile(sourceBucket: String, sourceObject: String, destinationBucket: String, destinationObject: String): Future[Option[StorageObject]] = {
+    implicit val service = GoogleInstrumentedService.Storage
+
+    //todo: if i decide to pass through the source StorageObject, make sure to reset any explicit ACLs on it otherwise it'll be wrong
+    val copier = getStorage(getBucketServiceAccountCredential).objects.copy(sourceBucket, sourceObject, destinationBucket, destinationObject, new StorageObject())
```

Comment:
this is the only thing im not sure of yet, need to look into it a little bit more. are there any strange side effects of passing in a new StorageObject instead of basing it off of the existing one?

Reply:
What's the purpose of the `StorageObject` parameter?

Reply:
you can set things like metadata and acls on it. i think using the blank storageobject is correct here because it will cleanse any of the metadata and alcs from the previous object